Acceptance tests require IOSXE_HOST environment variable despite IOSXE_URL fallback support

## Since version 0.10.0, acceptance tests fail when `IOSXE_URL` is set but `IOSXE_HOST` is not set, even though the provider code correctly supports using `IOSXE_URL` as a fallback. The test pre-check function unnecessarily enforces that `IOSXE_HOST` must be defined.

### Example:
```bash
(venv) user@host terraform-provider-iosxe % echo $IOSXE_URL
https://1.2.3.4

(venv) user@host terraform-provider-iosxe % echo $IOSXE_HOST


(venv) user@host terraform-provider-iosxe % echo $TF_ACC
1

(venv) user@host terraform-provider-iosxe % echo $C8000V
1
```

```bash
(venv) user@host terraform-provider-iosxe % terraform providers
Providers required by configuration:
.
└── module.iosxe
    ├── provider[registry.terraform.io/netascode/utils] 1.1.0-beta3
    ├── provider[registry.terraform.io/ciscodevnet/iosxe] 0.10.1
    └── module.model
        ├── provider[registry.terraform.io/netascode/utils] 1.1.0-beta3
        └── provider[registry.terraform.io/hashicorp/local] >= 2.5.2
```

```bash
(venv) user@host terraform-provider-iosxe % go test -v ./... -run TestAccIosxePIM
?       github.com/CiscoDevNet/terraform-provider-iosxe [no test files]
=== RUN   TestAccIosxePIM
    provider_test.go:47: IOSXE_HOST env variable must be set for acceptance tests
--- FAIL: TestAccIosxePIM (0.00s)
FAIL
FAIL    github.com/CiscoDevNet/terraform-provider-iosxe/internal/provider       0.406s
testing: warning: no tests to run
PASS
ok      github.com/CiscoDevNet/terraform-provider-iosxe/internal/provider/helpers       (cached) [no tests to run]
FAIL
```

### Explanation
The provider logic in `provider.go` (lines 289-319) is valid and correct however, the test pre-check logic in `provider_test.go` (lines 36-49) is overly restrictive and is missing the "fallback" behavior to support using the `IOSXE_URL` environment variable when the `IOSXE_HOST` environment variable is not defined.

### Available Workaround
The user **must** define the `IOSXE_HOST` environment variable with the host's URL prior to executing any acceptance tests.

```bash
(venv) user@host terraform-provider-iosxe % echo $IOSXE_URL

(venv) user@host terraform-provider-iosxe % echo $IOSXE_HOST
https://1.2.3.4
```

```bash
(venv) user@host terraform-provider-iosxe % go test -v ./... -run TestAccIosxePIM
?       github.com/CiscoDevNet/terraform-provider-iosxe [no test files]
=== RUN   TestAccIosxePIM
--- PASS: TestAccIosxePIM (10.26s)
PASS
ok      github.com/CiscoDevNet/terraform-provider-iosxe/internal/provider       10.673s
testing: warning: no tests to run
PASS
ok      github.com/CiscoDevNet/terraform-provider-iosxe/internal/provider/helpers       (cached) [no tests to run]
```

--- a/internal/provider/provider_test.go
+++ b/internal/provider/provider_test.go
@@ -44,6 +44,8 @@ func testAccPreCheck(t *testing.T) {
 		t.Fatal("IOSXE_PASSWORD env variable must be set for acceptance tests")
 	}
 	if v := os.Getenv("IOSXE_HOST"); v == "" {
-		t.Fatal("IOSXE_HOST env variable must be set for acceptance tests")
+		if v := os.Getenv("IOSXE_URL"); v == "" {
+			t.Fatal("IOSXE_HOST or IOSXE_URL env variable must be set for acceptance tests")
+		}
 	}
 }

